tests/nfa/builder.cc

```diff
@@ -141,3 +142,116 @@ TEST_CASE("parse_from_mata()") {
         }
     }
 }
+
+TEST_CASE("Create Tabakov-Vardi NFA") {
+    size_t num_of_states;
+    size_t alphabet_size;
+    float transition_density;
+    float final_state_density;
+
+    SECTION("EMPTY") {
+        num_of_states = 0;
+        alphabet_size = 0;
+        transition_density = 0;
+        final_state_density = 0;
+
+        Nfa nfa = mata::nfa::builder::create_random_nfa_tabakov_vardi(num_of_states, alphabet_size, transition_density, final_state_density);
+        CHECK(nfa.num_of_states() == 1);
+        CHECK(nfa.initial.size() == 1);
+        CHECK(nfa.final.size() == 1);
+        CHECK(nfa.delta.empty());
+    }
+
+    SECTION("10-5-0.5-0.5") {
+        num_of_states = 10;
+        alphabet_size = 5;
+        transition_density = 0.5;
```

Comment:
Is this correct? Did you want 0.5 or 5?

Reply:
Ok, I checked other tests, and you put larger numbers there, so this one probably tests a really small density, right?

Reply:
Yes, 0.5 is the correct value. I wanted to create *a normal* automaton with `10*0.5=5` transitions for each alphabet symbol.